core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -158,6 +160,21 @@ public Map<String, String> properties() {
     return properties;
   }
 
+  public int formatVersion() {
+    return formatVersion;
+  }
+
+  private int formatVersion(Table table) {
+    if (table instanceof HasTableOperations) {
+      return ((HasTableOperations) table).operations().current().formatVersion();
```

Comment:
Optional: What about an extra variable for `TableOperations`? We have an explicit casts + 3 method invocations, it may be easier to read with code on multiple lines and it would be consistent with the `metadataFileLocation` method.